### PR TITLE
Update pysam to 0.15

### PR DESCRIPTION
**Pull request by kyleabeauchamp**
* [ ] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [ ] This PR adds a new recipe.
* [ ] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [x] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

I am testing the bioconda build for pysam 0.15, as well as linked against libdeflate.

### CODE REVIEW

**Comment by kyleabeauchamp** [comment]
OK, there's something strange about the dependencies here in that when I didn't pin HTSlib to 1.9, it grabbed 1.7 for some reason (and erroring out during pysam compilation).  So I've chosen to pin to 1.9...

**Comment by kyleabeauchamp** [comment]
I've tested building the same branch locally with via docker/circleci and I can confirm (via `ldd`) that the SO file are linking to libdeflate.

**Comment by kyleabeauchamp** [comment]
Separately (https://github.com/samtools/htslib/pull/744), I've discovered that HTSLib links to both libdeflate AND (vanilla) zlib, so I've bumped the HTSLib recipe to include that explicit zlib dependence.

File: recipes/pysam/meta.yaml [hunk 1]
```diff
@@ -20,22 +20,24 @@ requirements:
   build:
     - {{ compiler('c') }}
   host:
-    - htslib
-    - samtools
-    - bcftools
+    - htslib 1.9
```
Comment (bgruening) [inline recipes/pysam/meta.yaml]: We should increase the version here: https://github.com/bioconda/bioconda-utils/blob/master/bioconda_utils/bioconda_utils-conda_build_config.yaml#L12

**Pull request merged**